### PR TITLE
Sort categorical fields to have same arrangement across rows

### PR DESCRIPTION
**Pull request by adarshlilha**
Added a configuration in muze config to sort both axes and facet fields.
If a field is not specified by user for sort, it is not sorted

```javascript
config({
    sort: {
       Origin: 'asc' (or 'desc'),
       Year: 'desc'
    }
})
```

### CODE REVIEW

File: packages/visual-group/src/encoder/cartesian-encoder.js [hunk 1]
```diff
@@ -121,9 +121,18 @@ export default class CartesianEncoder extends VisualEncoder {
                     fieldArr.forEach((field, axisIndex) => {
                         const key = !axisType ? `0${cIdx}${axisIndex}` : `${rIdx}0${axisIndex}`;
                         const dom = unitDomains[`${rIdx}${cIdx}`];
+                        const typeOfField = field.subtype();
+                        const sortingDetails = field.data()._sortingDetails;
```
Comment (ranajitbanerjee) [inline packages/visual-group/src/encoder/cartesian-encoder.js]: @adarshlilha Add an api in datamodel to get the sorting details

File: packages/visual-group/src/encoder/cartesian-encoder.js [hunk 2]
```diff
@@ -121,9 +121,18 @@ export default class CartesianEncoder extends VisualEncoder {
...
+                        const sortingDetailsFlattened = [].concat(...sortingDetails);
+
                         if (dom && Object.keys(dom).length !== 0) {
                             domains[axisType][key] = unionDomain([(domains[axisType] && domains[axisType][key]) || [],
-                                dom[`${field}`]], field.subtype());
+                                dom[`${field}`]], typeOfField);
+                            /* Sort categorical fields to ensure consistency across all rows
+                            only if field is categorical and is not explicitily sorted by user */
+                            if (typeOfField === CATEGORICAL && !sortingDetailsFlattened.includes(field.oneVar())) {
```
Comment (ranajitbanerjee) [inline packages/visual-group/src/encoder/cartesian-encoder.js]: Here sorting is happening multiple times for same field, which can be done only once.

Comment (adarshlilha) [reply packages/visual-group/src/encoder/cartesian-encoder.js]: Moved the sorting out

File: packages/visual-group/src/encoder/cartesian-encoder.js (hunk 1 above)
Comment (adarshlilha) [reply packages/visual-group/src/encoder/cartesian-encoder.js]: Added a method to fetch sortingDetails from the current field instance

**Comment by rousan** [comment]
Closing, as it is already merge into #206

**Pull request closed without merging**